openldap/map.jinja

```diff
@@ -1,97 +1,55 @@
-            '/etc/openldap/schema/inetorgperson.schema',
-        ],
-        'slapd_argsfile': '/var/run/openldap/slapd.args',
-        'slapd_pidfile': '/var/run/openldap/slapd.pid',
-        'slapd_directory': '/var/lib/ldap',
-        'slapd_modulepath': '/usr/lib64/openldap',
-        'slapd_modules': [
-            'back_mdb',
-            ],
-        'su_group': 'root',
-    },
-    'Suse': {
-        'client_config': '/etc/openldap/ldap.conf',
-        'client_pkg': 'openldap2-client',
-        'server_config': '/etc/openldap/slapd.conf',
-        'server_pkg': 'openldap2',
-        'service': 'slapd',
-        'slapd_includes': [
-            '/etc/openldap/schema/core.schema',
-            '/etc/openldap/schema/cosine.schema',
-            '/etc/openldap/schema/inetorgperson.schema',
-            '/etc/openldap/schema/rfc2307bis.schema',
-            '/etc/openldap/schema/yast.schema',
-         ],
-        'slapd_argsfile': '/var/run/openldap/slapd.args',
-        'slapd_pidfile': '/var/run/openldap/slapd.pid',
-        'slapd_directory': '/var/lib/ldap',
-        'slapd_modulepath': '/usr/lib64/openldap',
-        'slapd_modules': [
-            'back_mdb',
-         ],
-        'slapd_pidfile': '/var/run/slapd/slapd.pid',
-        'su_group': 'root',
-    },
-}, grain='os_family', merge=salt['pillar.get']('openldap:lookup')) %}
+# -*- coding: utf-8 -*-
+# vim: ft=jinja
+
+{#- Get the `tplroot` from `tpldir` #}
+{%- set tplroot = tpldir.split('/')[0] %}
+{#- Start imports as #}
+{%- import_yaml tplroot ~ "/defaults.yaml" as default_settings %}
+{%- import_yaml tplroot ~ "/osarchmap.yaml" as osarchmap %}
+{%- import_yaml tplroot ~ "/osfamilymap.yaml" as osfamilymap %}
+{%- import_yaml tplroot ~ "/osmap.yaml" as osmap %}
+{%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap %}
+
+{#- Retrieve the config dict only once #}
+{%- set _config = salt['config.get'](tplroot, default={}) %}
+
+{%- set defaults = salt['grains.filter_by'](
+      default_settings,
+      default=tplroot,
+      merge=salt['grains.filter_by'](
+        osarchmap,
+        grain='osarch',
+        merge=salt['grains.filter_by'](
+          osfamilymap,
+          grain='os_family',
+          merge=salt['grains.filter_by'](
+            osmap,
+            grain='os',
+            merge=salt['grains.filter_by'](
+              osfingermap,
+              grain='osfinger',
+              merge=salt['grains.filter_by'](
+                _config,
+                default='lookup'
+              )
+            )
+          )
+        )
+      )
+    )
+%}
+
+{%- set config = salt['grains.filter_by'](
+      {'defaults': defaults},
+      default='defaults',
+      merge=_config
+    )
+%}
+
+{%- set openldap = config %}
+
+{#- Post-processing for specific non-YAML customisations #}
+{%- if grains.os == 'MacOS' %}
+{%-   set macos_group = salt['cmd.run']("stat -f '%Sg' /dev/console") %}
+{%-   do TEMPLATE.update({'rootgroup': macos_group}) %}
+{%- endif %}
```

Comment:
If we're going to keep this section, then `TEMPLATE` needs to be updated accordingly.  However, there's no indication if this formula supports `MacOS` in the first place, so perhaps best to remove this block for the time being.

Reply:
done